cli/src/cmd_net.rs

```diff
@@ -0,0 +1,1475 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2024 Oxide Computer Company
+
+use std::{collections::HashMap, net::IpAddr};
+
+use crate::RunnableCmd;
+use anyhow::Result;
+use async_trait::async_trait;
+use clap::Parser;
+use colored::*;
+use futures::TryStreamExt;
+use oxide::{
+    context::Context,
+    types::{
+        Address, AddressConfig, BgpAnnounceSetCreate, BgpAnnouncementCreate, BgpPeer,
+        BgpPeerConfig, BgpPeerStatus, ImportExportPolicy, IpNet, LinkConfigCreate, LinkFec,
+        LinkSpeed, LldpServiceConfigCreate, Name, NameOrId, Route, RouteConfig,
+        SwitchInterfaceConfigCreate, SwitchInterfaceKind, SwitchInterfaceKind2, SwitchLocation,
+        SwitchPort, SwitchPortConfigCreate, SwitchPortGeometry, SwitchPortGeometry2,
```

Comment:
it's concerning to me that we've managed to have these types suffixed with 2 which indicates a name conflict, but that's a problem for another day. I should probably have some sort of lint check for that.